controllers/imagejob/imagejob_controller.go

```diff
@@ -102,62 +109,138 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
+			}
+
+			givenPodSpec := imageJob.Spec.JobTemplate.Spec
+			podSpec := v1.PodSpec{
+				RestartPolicy:      givenPodSpec.RestartPolicy,
+				ServiceAccountName: givenPodSpec.ServiceAccountName,
+				Containers:         []v1.Container{image},
+				NodeName:           nodeName,
+				Volumes:            []v1.Volume{{Name: runtimeName + "-sock-volume", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: mountPath}}}},
+			}
+
+			podName := image.Name + "-" + nodeName
+			pod := &v1.Pod{
+				TypeMeta: metav1.TypeMeta{},
+				Spec:     podSpec,
+				ObjectMeta: metav1.ObjectMeta{Namespace: "eraser-system",
+					Name:            podName,
+					Labels:          map[string]string{"name": image.Name},
+					OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(imageJob, imageJob.GroupVersionKind())}},
+			}
 
-		podName := image.Name + nodeName
-		pod := &v1.Pod{
-			TypeMeta:   metav1.TypeMeta{},
-			Spec:       podSpec,
-			ObjectMeta: metav1.ObjectMeta{Namespace: "eraser-system", Name: podName, Labels: map[string]string{"name": image.Name}},
+			// TODO: check if pod fits and can be scheduled on node
+			err = r.Create(ctx, pod)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
 		}
 
-		// TODO: check if pod fits and can be scheduled on node
-		err = r.Create(ctx, pod)
+	} else if imageJob.Status.Phase == eraserv1alpha1.PhaseRunning {
+		// get eraser pods
+		podList := &v1.PodList{}
+		err := r.List(ctx, podList, &client.ListOptions{
+			Namespace:     "eraser-system",
+			LabelSelector: labels.SelectorFromSet(map[string]string{"name": imageJob.Spec.JobTemplate.Spec.Containers[0].Name})})
 		if err != nil {
-			return ctrl.Result{}, err
+			log.Println(err)
 		}
-		controllerLog.Info("created pod", "name", podName, "node", nodeName, "podType", image.Name)
 
+		failed := 0
+		success := 0
+
+		// if all pods are complete, job is complete
+		if podsComplete(podList.Items) {
+			// get status of pods
+			for _, p := range podList.Items {
+				if p.Status.Phase == v1.PodSucceeded {
+					success++
+				} else {
+					failed++
+				}
+			}
+
+			imageJob.Status = eraserv1alpha1.ImageJobStatus{
+				Desired:   imageJob.Status.Desired,
+				Succeeded: success,
+				Failed:    failed,
+				Phase:     eraserv1alpha1.PhaseCompleted,
+			}
+
+			updateJobStatus(*imageJob)
+
+			// transfer results from imageStatus objects to imageList
+			statusList := &eraserv1alpha1.ImageStatusList{}
+			err = r.List(ctx, statusList, &client.ListOptions{
+				Namespace: "eraser-system"})
+			if err != nil {
+				log.Println(err)
+			}
+
+			var nodeResult []eraserv1alpha1.NodeResult
+
+			for _, s := range statusList.Items {
+				nodeResult = append(nodeResult, eraserv1alpha1.NodeResult{
+					Name:   s.Result.Node,
+					Images: s.Result.Results,
+				})
+			}
+
+			imageList := &eraserv1alpha1.ImageListList{}
+			err = r.List(ctx, imageList)
+			if err != nil {
+				log.Println(err)
+			}
+
+			for _, l := range imageList.Items {
```

Comment:
are we updating all imagelists here?

Reply:
it should've only been the current one. Just changed it so imagejob receives imagelist name and uses that to get the list